solc/CommandLineParser.h

```diff
@@ -0,0 +1,209 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Validates and parses command-line options into an internal representation.
+ */
+#pragma once
+
+#include <libsolidity/interface/CompilerStack.h>
+#include <libsolidity/interface/DebugSettings.h>
+#include <libsolidity/interface/FileReader.h>
+#include <libsolidity/interface/ImportRemapper.h>
+#include <libyul/AssemblyStack.h>
+#include <liblangutil/EVMVersion.h>
+
+#include <boost/program_options.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace solidity::frontend
+{
+
+enum class InputMode
+{
+	Compiler,
+	CompilerWithASTImport,
+	StandardJson,
+	Linker,
+	Assembler,
+};
+
+struct OutputSelection
+{
+	bool operator!=(OutputSelection const& _other) const { return !(*this == _other); }
+	bool operator==(OutputSelection const& _other) const;
```

Comment:
Maybe both `noexcept`? do we have a standing on the use of that modifier?

Reply:
I think we do not. Personally, I don't mind it but I just keep forgetting about it because in normal circumstances it has no consequences. I'll try to be more consistent about it.